### ISSUE TITLE
Enable Goma on more build steps

### ISSUE THREAD

**Issue by mihaiplesa** [brave/brave-browser#16659]
For example network audit or anything else relevant.

**Comment by mariospr** [comment]
I believe the only three cases that can benefit from GOMA builds are these:
  1. `npm run build` -- builds the `brave` binary. Already covered :heavy_check_mark: 
  2. `npm run test <suite>` -- builds C++ files necesary to build `<suite>` -- we tried to add support for this in https://github.com/brave/brave-core/pull/8457 but then it got reverted in https://github.com/brave/brave-core/pull/9398 because it seemed to be too slow (maybe that has changed now, though?)
  3. `npm run network-audit` -- builds the `brave_network_audit_tests` binary and runs it with special timeouts. It's really an alias to a `npm run test` command running underneath, so supporting (2) would support this (3) automatically (as well as running upstream tests, which we're trying to enable in https://github.com/brave/brave-core/pull/11814).

In other words, the **only build step that would need adding GOMA support for is `npm run test`**, so maybe it would be worth trying https://github.com/brave/brave-core/pull/8457 once again and merging it if it's not as slow as before. If it is still slow, we'd need to spend some time trying to figure out why and how to solve it, but I'd say re-trying https://github.com/brave/brave-core/pull/8457 now that the GOMA infra has changed quite a bit (way more nodes!) might be a good start.

@mihaiplesa WDYT?

**Comment by mihaiplesa** [comment]
@mariospr `create_dist` also supports Goma. Added for network audit just now.
